tests/metrics.go

```diff
@@ -37,19 +39,45 @@ func GetNodesMetrics(ctx context.Context, nodeURIs []string) (NodesMetrics, erro
 	return metrics, nil
 }
 
-func GetFirstMetricValue(metrics NodeMetrics, name string) (float64, bool) {
+// GetMetricValue returns the value of the specified metric which has the
+// required labels.
+//
+// If multiple metrics match the provided labels, the first metric found is
+// returned.
+//
+// Only Counter and Gauge metrics are supported.
+func GetMetricValue(metrics NodeMetrics, name string, labels prometheus.Labels) (float64, bool) {
 	metricFamily, ok := metrics[name]
-	if !ok || len(metricFamily.Metric) < 1 {
+	if !ok {
 		return 0, false
 	}
 
-	metric := metricFamily.Metric[0]
-	switch {
-	case metric.Gauge != nil:
-		return metric.Gauge.GetValue(), true
-	case metric.Counter != nil:
-		return metric.Counter.GetValue(), true
-	default:
-		return 0, false
+	for _, metric := range metricFamily.Metric {
+		if !labelsMatch(metric, labels) {
+			continue
+		}
+
+		switch {
+		case metric.Gauge != nil:
+			return metric.Gauge.GetValue(), true
+		case metric.Counter != nil:
+			return metric.Counter.GetValue(), true
+		}
+	}
+	return 0, false
+}
+
+func labelsMatch(metric *dto.Metric, labels prometheus.Labels) bool {
+	var found int
+	for _, label := range metric.Label {
+		expectedValue, ok := labels[label.GetName()]
+		if !ok {
+			continue
```

Comment:
no point in `continue` here; you can just 
```golang
return false
```

Reply:
also, comparing `len(metric.Label) == len(labels)` before the loop could save you the need to track `found`.

Reply:
This wasn't quite what I had intended for this function. Perhaps we should improve the documentation. If we have metrics like:
```
{a:"0", b:"1"}
{a:"2", b:"3"}
```
I intended to be able to call this function with `{a:"0"}` and have it return the first metric (even though the specified labels didn't mention anything about `b`).

Reply:
We could also just expect the labels to be fully specified... 🤷

Reply:
ohh - that's create even bigger issue ;-(
the `prometheus.Labels` is a map, so iteration on that would not have a consistent ordering..

Reply:
I'm not sure I follow. This function isn't iterating over a map.

Reply:
oh - sorry. yes, it was very confusing, but you're correct. it should be ok.